bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/WidgetLabelSource.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.rest.sitemap.internal;
+
+import org.openhab.core.model.sitemap.sitemap.Widget;
+
+/**
+ * @author Danny Baumann - Initial contribution
+ */
+public enum WidgetLabelSource {
+    SitemapDefinition,
+    ItemLabel,
+    None;
+
+    public static WidgetLabelSource forWidget(Widget w, String label) {
+        if (label == null || label.isEmpty()) {
+            return None;
+        } else if (w.getLabel() != null) {
+            return SitemapDefinition;
+        } else {
+            return ItemLabel;
+        }
+    }
```

Comment:
In case the label is set to "[%s]" for example, either on sitemap element or on item, your method will return SitemapDefinition or ItemLabel, but not None. Is it really what you expect ?

Reply:
I think so. After all, there *is* a label to work with, even if it almost only consists of the item state. I introduced `NONE` only because if there's no label at all, neither `ITEM` nor `SITEMAP` do really fit.